core/metrics/meter_wrapper.cxx

```diff
@@ -0,0 +1,184 @@
+/* -*- Mode: C++; tab-width: 4; c-basic-offset: 4; indent-tabs-mode: nil -*- */
+/*
+ *   Copyright 2024. Couchbase, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+#include "meter_wrapper.hxx"
+
+#include <couchbase/error_codes.hxx>
+
+namespace couchbase::core::metrics
+{
+namespace
+{
+auto
+snake_case_to_camel_case(const std::string& s) -> std::string
+{
+  std::string res{};
+  bool capitalize{ true }; // First letter should be capitalized
+  for (const char c : s) {
+    if (c == '_') {
+      capitalize = true; // Next letter should be capitalized
+      continue;
+    }
+    if (capitalize) {
+      res += static_cast<char>(std::toupper(c));
+      capitalize = false;
+    } else {
+      res += c;
+    }
+  }
+  return res;
+}
+
+auto
+extract_error_name(std::error_code ec) -> std::string
+{
+  const std::string::size_type pos = ec.message().find(' ');
+  if (pos != std::string::npos) {
+    return ec.message().substr(0, pos);
+  }
+  return ec.message();
+}
+
+auto
+service_to_string(service_type s) -> std::string
```

Comment:
it will create new string every time. It should return `const std::string&`, `const char *` or at least `std::string_view`

Reply:
Won't we have to eventually create a copy of the string anyway, because we have to insert it in the tags map that meters expect as `std::map<std::string, std::string>`?